### DISCUSSION TITLE
Iobroker integration

### DISCUSSION THREAD

**Discussion by Evilghost12** [revk/xESP32-Faikout#202] [Q&A]
Has anyone already tried to integrate this WiFi module into Iobroker via MQTT? Unfortunately, this module does not create automatic objects. Unfortunately I have little experience with JSON in IoBroker.

**Comment by openKimchi** [comment]
ioBroker is more popular in German-speaking countries. Are you German?
Maybe I can help you. However, you should have a little knowledge of MQTT and also how to create your own data points and scripts.

**Comment by Evilghost12** [comment]
YES, I'm from Austria :-) ; In the meantime I have read a little and created the data points that I need and I also read out the JSON attributes and create data points with them - it seems to work - THANK YOU very much

**Reply by revk** [reply]
Glad to hear it.

There is a wiki on this page if you want to write it up for others?

**Reply by Evilghost12** [reply]
Yes, I can do that - when I have finished my project - I still have to buy and create 7 more units -> Only in the temperature values do I occasionally get values that look like this [20,20.3,20.5] -> I don't know how I should handle the values in order to display a normal temperature.

**Reply by revk** [reply]
There is a setting to force single values, that is a min/ave/max array.

**Reply by Evilghost12** [reply]
fixstatus ?

**Reply by revk** [reply]
The manual should explain but sounds like it

**Reply by Evilghost12** [reply]
Thx Very Much

**Reply by slackfish** [reply]
Can you share how you did it?

**Reply by Evilghost12** [reply]
MQTT Protocol is documented. You have to create the objects you need to send an command yourself. Like here in this structure. 
![image](https://github.com/revk/ESP32-Faikin/assets/75432774/6d1c1471-beb7-467b-a886-d3f0d23142dd)

**Reply by Evilghost12** [reply]
"OG-Schlafzimmer" is the Device Name.

**Reply by Evilghost12** [reply]
You get the status of the device here as JSON - you break it down in a script and write it into your own user objects.
![image](https://github.com/revk/ESP32-Faikin/assets/75432774/2377590b-a11a-4783-899a-308eaeceaf80)
![image](https://github.com/revk/ESP32-Faikin/assets/75432774/fe98b478-aed7-4ff7-a3df-3958908159f6)

**Reply by Evilghost12** [reply]
It's still too early for me to give precise instructions - as it's not perfect yet.

**Reply by slackfish** [reply]
Thanks for the insight i figured out parsing the state into iobroker datapoints, but not sure how to send it to the Faikinmodule yet.

**Reply by slackfish** [reply]
So I was able to change the temperature via mqtt msgs. But it seems all commands with an empty payload do not work. I also tried sending a JSON via the control command but that does not work either.

**Reply by revk** [reply]
Some specific examples would be helpful.

**Reply by slackfish** [reply]
Maybe im missing somethin if i send this payload via MQTT:
{"power":"true", "fan":"5", "mode":"C"}
The Fan changes to 5 the mode changes to cooling but it stays off. Is there some option which blocks powering on via mqtt?

**Reply by revk** [reply]
Err, surely `"power":true` not `"power":"true"` ?

**Reply by slackfish** [reply]
Well that did it. Thansk alot. I feel stupid now.  :-)

But still the "command/Faikin/on" or "command/Faikin/heat" do not work for me I tried setting payload to Null,"" or without.

**Reply by fichtenmoped82** [reply]
This is how it works for me with IOBroker using JScript:

//faikinog is the hostname I gave the faikin device in its webgui.
sendTo('mqtt.0', 'sendMessage2Client', {"topic": 'command/faikinog/on', "message": ''});  //switching on

I'm using the MQTT Broker/Client in version 5.2.0.0 just as a broker.

I'm reading states using: 

const obj_og = JSON.parse(getState('mqtt.0.state.faikinog.status').val);

log(obj_og); //logs the containing state infos 

Hope this helps other beginners to find a starting point quicker :)

**Comment by slackfish** [comment]
This is the nodered flow im trying rn.
@revk 
`[
    {
        "id": "bb8d46005f9045e9",
        "type": "tab",
        "label": "Flow 2",
        "disabled": false,
        "info": "",
        "env": []
    },
    {
        "id": "351419d0ce8e14fa",
        "type": "mqtt out",
        "z": "bb8d46005f9045e9",
        "name": "MQTT",
        "topic": "",
        "qos": "",
        "retain": "",
        "respTopic": "",
        "contentType": "",
        "userProps": "",
        "correl": "",
        "expiry": "",
        "broker": "8d8f0453e02a2bf7",
        "x": 1450,
        "y": 380,
        "wires": []
    },
    {
        "id": "0f90dbf3e925e46c",
        "type": "inject",
        "z": "bb8d46005f9045e9",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "setting/Faikin-Eltern",
        "payload": "{\"power\":\"true\"}",
        "payloadType": "json",
        "x": 1000,
        "y": 320,
        "wires": [
            [
                "351419d0ce8e14fa"
            ]
        ]
    },
    {
        "id": "b91eec9a16d9b1e4",
        "type": "inject",
        "z": "bb8d46005f9045e9",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "command/Faikin-Eltern/control",
        "payload": "{\"power\":\"true\", \"fan\":\"5\", \"mode\":\"C\"}",
        "payloadType": "json",
        "x": 910,
        "y": 400,
        "wires": [
            [
                "351419d0ce8e14fa"
            ]
        ]
    },
    {
        "id": "5a52be043f757f5e",
        "type": "inject",
        "z": "bb8d46005f9045e9",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "command/Faikin-Eltern/temp",
        "payload": "25",
        "payloadType": "num",
        "x": 980,
        "y": 460,
        "wires": [
            [
                "351419d0ce8e14fa"
            ]
        ]
    },
    {
        "id": "18a7ae12492f1999",
        "type": "inject",
        "z": "bb8d46005f9045e9",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "command/Faikin-Eltern/temp",
        "payload": "17",
        "payloadType": "num",
        "x": 980,
        "y": 520,
        "wires": [
            [
                "351419d0ce8e14fa"
            ]
        ]
    },
    {
        "id": "1ec0f88e6c74d1da",
        "type": "inject",
        "z": "bb8d46005f9045e9",
        "name": "",
        "props": [
            {
                "p": "topic",
                "vt": "str"
            },
            {
                "p": "payload"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "command/Faikin-Eltern/heat",
        "payload": "[]",
        "payloadType": "bin",
        "x": 910,
        "y": 580,
        "wires": [
            [
                "351419d0ce8e14fa"
            ]
        ]
    },
    {
        "id": "8d8f0453e02a2bf7",
        "type": "mqtt-broker",
        "name": "Iobroker",
        "broker": "iobroker.rudel",
        "port": "1986",
        "clientid": "",
        "autoConnect": true,
        "usetls": false,
        "protocolVersion": "4",
        "keepalive": "60",
        "cleansession": true,
        "autoUnsubscribe": true,
        "birthTopic": "",
        "birthQos": "0",
        "birthRetain": "false",
        "birthPayload": "",
        "birthMsg": {},
        "closeTopic": "",
        "closeQos": "0",
        "closeRetain": "false",
        "closePayload": "",
        "closeMsg": {},
        "willTopic": "",
        "willQos": "0",
        "willRetain": "false",
        "willPayload": "",
        "willMsg": {},
        "userProps": "",
        "sessionExpiry": ""
    }
]`

**Reply by slackfish** [reply]
P.S. The webcontrols all work flawlessly.
